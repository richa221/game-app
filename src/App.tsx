import React, { Component, Fragment } from 'react';
import Canvas from './components/canvas';
import NewGame from "./components/newGame";
import EndGame from "./components/endGame";
import CurrentCount from "./components/currentCount";

interface AppProps {
   //code related to your props goes here
   createBoard(): void;
   endBoard(): void;
   gameState: any;
   squareClicked: any;
}

class App extends Component<AppProps> {
  componentDidMount() {   

    const canvas:any = document.getElementById("color-game-app");
    canvas.style.width = `${ canvas.style.height }px`;

    this.props.createBoard();
    this.props.endBoard();
  }

  render() {
    return (
      <Fragment>
        <div id={ "content" }>
          <div className={ "gridContainer" }>
            <div className={ "title" }>
              React Game App
            </div>
            <div className={ "informationContainer" }>
              <div className={ "informationLeft" }>
                <NewGame createBoard={ this.props.createBoard }/>
              </div>
             
              <div className={ "informationRight" }>
                <CurrentCount gameState={ this.props.gameState }/>
              </div>
              <br/>
              <br/>
              <br/>
              <br/>
               <div className={ "informationRight" }>
                <EndGame endBoard={ this.props.endBoard }/>
              </div>
            </div>
            <div className={ "game" }>
              <Canvas
                gameState={ this.props.gameState }
                squareClicked={ this.props.squareClicked }
                createBoard={ this.props.createBoard }
                endBoard={ this.props.endBoard }
              />
            </div>
            <div className={ "settings" }></div>
          </div>
        </div>
      </Fragment>
    );
  }
}

export default App;
