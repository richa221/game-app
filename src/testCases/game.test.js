import App from '../App';
import Adapter from 'enzyme-adapter-react-16'
import { configure,shallow } from 'enzyme'
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore({});
const mockCallBack = jest.fn(); 
configure({ adapter: new Adapter() })

const gameContainerElement = shallow((<Provider store={store}><App /></Provider>));

describe('Game container snapshot testing',()=>{
  it('renders app module correctly',()=>{
      expect(gameContainerElement).toMatchSnapshot();
  });
});
