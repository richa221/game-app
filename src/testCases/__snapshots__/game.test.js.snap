// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game container snapshot testing renders app module correctly 1`] = `ShallowWrapper {}`;
